### ISSUE TITLE
The repo appears rather large? Maybe something should be added to .gitignore?

### ISSUE THREAD

**Issue by maacl** [plul/slime-mold-simulation#1]
<img width="528" alt="image" src="https://github.com/plul/slime-mold-simulation/assets/64092/612289b3-d023-4cbf-b794-15fd6fb1a08f">

**Comment by plul** [comment]
I think it's just the videos used in the presentation probably.

**Comment by maacl** [comment]
Cloning fails for me with the below:

Cloning into 'slime-mold-simulation'...
remote: Enumerating objects: 3457, done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL
(err 8)
error: 7634 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output

**Comment by plul** [comment]
That's strange, probably your connection or github. 

There is ~50 MB worth of images+video used in the presentation, and another ~50 MB worth of vendored node_modules for frontend dependencies, but I can still clone it in about ~5 seconds on my connection.

**Comment by maacl** [comment]
Must have been a connection issue. Works now.

**Comment by maacl** [comment]
I am getting the below error when trying to run the simulation. I am on MacOS.

`cargo run --manifest-path crates/simulation/Cargo.toml -- --resolution  "1000x1000"
warning: unused variable: `tcp_listener_join_handle`
   --> src/main.rs:117:9
    |
117 |     let tcp_listener_join_handle = tokio_runtime.spawn({
    |         ^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_tcp_listener_join_handle`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `user_input_handling_thread`
   --> src/main.rs:137:9
    |
137 |     let user_input_handling_thread = {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_user_input_handling_thread`

warning: unused variable: `sync_shader_parameters_task`
   --> src/main.rs:215:9
    |
215 |     let sync_shader_parameters_task: tokio::task::JoinHandle<_> = {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_sync_shader_parameters_task`

warning: unused variable: `simulation_driver_thread`
   --> src/main.rs:228:9
    |
228 |     let simulation_driver_thread = {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_simulation_driver_thread`

warning: function `random_number` is never used
   --> src/main.rs:398:4
    |
398 | fn random_number(from: f32, to: f32) -> f32 {
    |    ^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: function `scale` is never used
   --> src/main.rs:417:4
    |
417 | fn scale(vec: [f32; 2], factor: f32) -> [f32; 2] {
    |    ^^^^^

warning: `simulation` (bin "simulation") generated 6 warnings (run `cargo fix --bin "simulation"` to apply 4 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `crates/simulation/target/debug/simulation --resolution 1000x1000`
thread 'main' panicked at src/main.rs:442:5:
assertion failed: surface_caps.formats.contains(&texture_format)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
libc++abi: terminating due to uncaught foreign exception
zsh: abort      cargo run --manifest-path crates/simulation/Cargo.toml -- --resolution`

**Comment by plul** [comment]
Yeah okay. At some point a texture format must be chosen for the window surface during setup.
I hardcoded the choice here: https://github.com/plul/slime-mold-simulation/blob/master/crates/simulation/src/main.rs#L750

This works for all my hardware but evidently not for yours. 
Ideally the code would dynamically select a texture format that is supported by the hardware. 
You can try changing to a different one and see what your hardware supports with 
```rust
dbg!(surface_caps);
```

at this point: https://github.com/plul/slime-mold-simulation/blob/master/crates/simulation/src/main.rs#L441

**Comment by maacl** [comment]
Thanks. I get the capabilities bellow. I have tried with various combinations of formats and alphamodes (L 453), but they all give me just an empty window. I don't see any errors anywhere.

`[src/main.rs:442:5] &surface_caps = SurfaceCapabilities {
    formats: [
        Bgra8UnormSrgb,
        Bgra8Unorm,
        Rgba16Float,
        Rgb10a2Unorm,
    ],
    present_modes: [
        Fifo,
        Immediate,
    ],
    alpha_modes: [
        Opaque,
        PostMultiplied,
    ],
    usages: TextureUsages(
        COPY_SRC | COPY_DST | RENDER_ATTACHMENT,
    ),
}`

**Comment by plul** [comment]
Hm hard to say what may be wrong. 
If you feel like debugging it, I would try putting out a solid color from the fragment shader other than black, just to see if it is the rendering that doesn't work or the simulation.

**Comment by maacl** [comment]
The window is all white, so it would appear that it is the rendering. Also the process is running 19 threads and using approx 6% CPU and approx 10 % GPU.

<img width="1040" alt="image" src="https://github.com/plul/slime-mold-simulation/assets/64092/c6839a2a-4f9c-426c-8f01-5ceace70b8ad">
